src/pages/ballot/confirmation.tsx

```diff
@@ -1,17 +1,29 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+
 import { useBallot } from "~/contexts/Ballot";
 import { BallotConfirmation } from "~/features/ballot/components/BallotConfirmation";
 import { Layout } from "~/layouts/DefaultLayout";
 
 export default function BallotConfirmationPage() {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
   const { ballot } = useBallot();
+  const router = useRouter();
+
+  useEffect(() => {
+    if (ballot?.published) {
+      setIsLoading(false);
+    }
 
-  if (!ballot) return null;
+    if (ballot && !ballot.published) {
+      void router.push("/ballot");
```

Comment:
Why do you need void here?

Reply:
since the `router.push` function is actually an async function, so I just copy paste the code here: https://github.com/privacy-scaling-explorations/maci-rpgf/blob/da5b785c14fb0fb6ffb4bd34c47f8939aa944d8c/src/layouts/BaseLayout.tsx#L51

should both be modified by removing that `void`?

Reply:
Yes, let's use await instead.